includes/Checker/Checks/Plugin_Repo/Plugin_Header_Fields_Check.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * Class Plugin_Header_Fields_Check.
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks\Plugin_Repo;
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Check_Categories;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Static_Check;
+use WordPress\Plugin_Check\Traits\Amend_Check_Result;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+/**
+ * Check for plugin header fields.
+ *
+ * @since 1.2.0
+ */
+class Plugin_Header_Fields_Check implements Static_Check {
+
+	use Amend_Check_Result;
+	use Stable_Check;
+
+	/**
+	 * Gets the categories for the check.
+	 *
+	 * Every check must have at least one category.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @return array The categories for the check.
+	 */
+	public function get_categories() {
+		return array( Check_Categories::CATEGORY_PLUGIN_REPO );
+	}
+
+	/**
+	 * Amends the given result by running the check on the associated plugin.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of the check).
+	 *
+	 * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 */
+	public function run( Check_Result $result ) {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_main_file = $result->plugin()->main_file();
+		$plugin_header    = get_plugin_data( $plugin_main_file );
+
+		$labels = array(
+			'Name'            => 'Plugin Name',
+			'PluginURI'       => 'Plugin URI',
+			'Version'         => 'Version',
+			'Description'     => 'Description',
+			'Author'          => 'Author',
+			'AuthorURI'       => 'Author URI',
+			'TextDomain'      => 'Text Domain',
+			'DomainPath'      => 'Domain Path',
+			'Network'         => 'Network',
```

Comment:
This may be out of scope, but we could add extra validation for `Network` as well. 
Can only be set to `true`, and should be left out when not needed.

Reply:
With get_plugin_data(), this will not be possible as `Network` value comes as `true` or `false` from this function. It does not return actual header value. `$plugin_data['Network'] = ( 'true' === strtolower( $plugin_data['Network'] ) );`